matrix-api.c

```diff
@@ -431,31 +434,42 @@ static gchar *_build_request(PurpleAccount *acct, const gchar *url,
     if(body != NULL)
         g_string_append(request_str, body);
 
+    if(extra_data != NULL)
+        g_string_append_len(request_str, extra_data, extra_len);
+
+    *total_len = request_str->len; /* TODO: Surely there must be a function!? */
```

Comment:
since _build_request is only called in one place, it might be preferable to return the GString * rather than have to add the total_len arg